### PR TITLE
Mocha tests for BinaryWriter, Catalog.

### PR DESCRIPTION
**Pull request by zslayton**
Follow-on work from #452.

*Description of changes:*

Mocha tests for `BinaryWriter` and `Catalog`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/IonBinaryWriter.ts [hunk 1]
```diff
@@ -221,7 +221,7 @@ export class BinaryWriter extends AbstractWriter {
 
   writeSymbol(value: string) : void {
     this.checkWriteValue();
-    if (value === null) {
+    if (value === null || value === undefined) {
```
Comment (wesboyt) [inline src/IonBinaryWriter.ts]: do we want to allow undefined input here?

File: test/IonBinaryWriter.ts [hunk 2]
```diff
@@ -0,0 +1,1139 @@
...
+        expected: [0x10]
+    },
+    {
+        name: "Writes null boolean by detecting null",
+        instructions: (writer) => writer.writeBoolean(null),
+        expected: [0x1f]
+    },
+    {
+        name: "Writes null boolean by detecting undefined",
+        instructions: (writer) => writer.writeBoolean(undefined),
```
Comment (wesboyt) [inline test/IonBinaryWriter.ts]: should we open an issue here tracking removal undefined from the boolean api?

File: src/IonBinaryWriter.ts (hunk 1 above)
Comment (zslayton) [reply src/IonBinaryWriter.ts]: > do we want to allow undefined input here? I personally would rather throw.

Me too. I've opened [an issue](https://github.com/amzn/ion-js/issues/456) to discuss and track this change.

In the meantime, `writeSymbol` is the only odd-man-out that doesn't accept `undefined`. This required a test to be skipped, so I addressed it here. We can change the behavior across the board all in a future PR.

File: test/IonBinaryWriter.ts (hunk 2 above)
Comment (zslayton) [reply test/IonBinaryWriter.ts]: #456 covers this.

**Pull request merged**